interfaces/builtin/basedeclaration.go

```diff
@@ -34,100 +34,258 @@ series: 16
+    allow-installation:
+      slot-snap-type:
+        - core
+  mir:
+    allow-installation:
+      slot-snap-type:
+        - app
   modem-manager:
     deny-auto-connection: true
   mount-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   mpris:
-    deny-auto-connection: true
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-auto-connection: true
+  network:
+    allow-installation:
+      slot-snap-type:
+        - core
+  network-bind:
+    allow-installation:
+      slot-snap-type:
+        - core
   network-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   network-manager:
     deny-auto-connection: true
   network-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   network-setup-observe:
-    deny-auto-connection: true
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+  opengl:
+    allow-installation:
+      slot-snap-type:
+        - core
+  optical-drive:
+    allow-installation:
+      slot-snap-type:
+        - core
   ppp:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   process-control:
-    deny-auto-connection: true
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+  pulseaudio:
+    allow-installation:
+      slot-snap-type:
+        - core
   removable-media:
-    deny-auto-connection: true
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+  screen-inhibit-control:
+    allow-installation:
+      slot-snap-type:
+        - core
   serial-port:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
     deny-auto-connection: true
   snapd-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: false
   system-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   system-trace:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   time-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   timeserver-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   timezone-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   tpm:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   udisks2:
     deny-auto-connection: true
+  unity7:
+    allow-installation:
+      slot-snap-type:
+        - core
+  upower-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
+  x11:
+    allow-installation:
+      slot-snap-type:
+        - core
 `
```

Comment:
This is all correct (except docker, see above) for while we don't have snap declarations.